Assets/Scripts/UI/GalleryMenu.cs

```diff
@@ -0,0 +1,221 @@
+
+                gun.transform.localScale = new Vector3(scaleFactor / gun.transform.lossyScale.x, scaleFactor / gun.transform.lossyScale.y, scaleFactor / gun.transform.lossyScale.z);
+
+                // Recalculate bounds
+                foreach (var renderer in gun.GetComponentsInChildren<Renderer>())
+                {
+                    renderer.gameObject.layer = LayerMask.NameToLayer("UI");
+                    bounds.Encapsulate(renderer.bounds);
+                }
+
+                // Center the gun on the grid cell
+                gun.transform.localPosition = new Vector3(0 - bounds.center.z * gridCellSize.x / 4, gridCellSize.y / 3, -bounds.extents.x);
+
+                // Rotate the weapon to the correct angle
+                gun.transform.Rotate(Vector3.up * 90);
+
+                // Add the weapon name
+                string name = GunFactory.GetGunName(weapon.body, weapon.barrel, weapon.extension);
+                gun.name = name;
+                gridElement.GetComponentInChildren<TMP_Text>().text = name;
+            } else
+            {
+                // We are out of displayable weapons, disable objects
+                gridElement.gameObject.SetActive(false);
+            }    
+        }
+    }
+
+    public void NextPage(InputAction.CallbackContext ctx)
+    {
+        if(pageIndex == maxPages - 1)
+        {
+            pageIndex = 0;
+        }
+        else
+        {
+            pageIndex++;
+        }
+
+        selectedTab.GetComponent<Image>().color = disabledColor;
+        selectedTab = tabs[pageIndex];
+        selectedTab.GetComponent<Image>().color = selectedColor;
+
+        PopulateGrid(pageIndex);
+    }
+
+    public void PrevPage(InputAction.CallbackContext ctx)
+    {
+        if (pageIndex == 0)
+        {
+            pageIndex = maxPages - 1;
+        }
+        else
+        {
+            pageIndex--;
+        }
+
+        selectedTab.GetComponent<Image>().color = disabledColor;
+        selectedTab = tabs[pageIndex];
+        selectedTab.GetComponent<Image>().color = selectedColor;
+
+        PopulateGrid(pageIndex);
+    }
+
+    public void Back(InputAction.CallbackContext ctx)
+    {
+        mainMenuController.ReturnToMainMenu();
+    }
+
+    /// <summary>
+    /// Creates and returns an array of all possible weapon combinations
+    /// </summary>
+    private Weapon[] CreateAllWeapons()
+    {
+        StaticInfo staticInfo = StaticInfo.Singleton;
+
+        List<Weapon> weaponList = new List<Weapon>();
+        for(int body = 0; body < staticInfo.Bodies.Count; body++)
+        {
+            for(int barrel = 0; barrel < staticInfo.Barrels.Count; barrel++)
+            {
+                for(int extension = -1; extension < staticInfo.Extensions.Count; extension++)
+                {
+                    if(extension == -1)
+                    {
+                        weaponList.Add(new Weapon(staticInfo.Bodies[body], staticInfo.Barrels[barrel]));
```

Comment:
You could do this extension-less addition _before_ the extension loop, having the loop start at `-1` and then not using the loop variable is a tad too confusing.